File: 0066-plus-one/solution.js
/**
 * @param {number[]} digits
 * @return {number[]}
 */
var plusOne = function(digits) {
  let i = digits.length;

  digits[i-1]++;
  
  while(i-- > 0) {
    if (digits[i] < 10) break;
    digits[i] = 0;
    if (i > 0) digits[i-1]++;
    else digits.unshift(1);
    
  }

  return digits;
};

File: 0066-plus-one/solution.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        d = len(digits) - 1
        while d >= 0:
            if digits[d] == 9:
                digits[d] = 0
                d -= 1
            else:
                digits[d] += 1
                return digits
        return [1] + digits

